### PR TITLE
Refs #21204 - default PXE entry from settings

### PR DESCRIPTION
**Pull request by lzap**
This unifies how default entry is being selected via new setting helper function. Note grub1 only supports number, there is a conversion. Method "index" should be allowed by safemode. For grub2 labels are created using "id" option:

https://www.gnu.org/software/grub/manual/grub/grub.html#default

See https://github.com/theforeman/foreman/pull/4891 for more info.

### CODE REVIEW

**Comment by lzap** [comment]
Mis-pushed an old commit, rebased. Do not merge until the core PR is merged of course :-)

**Comment by mmoll** [comment]
please rebase :)

**Comment by lzap** [comment]
Will do once core is merged.

File: provisioning_templates/PXEGrub2/pxegrub2_default_local_boot.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@ name: PXEGrub2 default local boot
 %>
 <%# Used to boot provisioned hosts, do not associate or change the name. %>
 
-set default=0
+set default=<%= global_setting("default_pxe_item_local", "local_chain_hd0") %>
```
Comment (iNecas) [inline provisioning_templates/PXEGrub2/pxegrub2_default_local_boot.erb:8]: Shouldn't we use the `index` here as well?

File: provisioning_templates/PXEGrub/pxegrub_global_default.erb [hunk 2]
```diff
@@ -3,8 +3,12 @@ kind: PXEGrub
 name: PXEGrub global default
 %>
 <%# Used to boot unknown hosts, do not associate or change the name. %>
-
-default=0
+<%
+  # Grub1 only supports numeric default statement, this array allows conversion to number
+  items = ["local", "discovery"]
+  default_item = items.index(global_setting("default_pxe_item_global", "local"))
```
Comment (iNecas) [inline provisioning_templates/PXEGrub/pxegrub_global_default.erb]: What happens, if the item is not found? Shouldn't we raise some error with explanation what went wrong?

File: provisioning_templates/PXEGrub2/pxegrub2_global_default.erb [hunk 3]
```diff
@@ -4,7 +4,7 @@ name: PXEGrub2 global default
 %>
 <%# Used to boot unknown hosts, do not associate or change the name. %>
 
-default=0
+default=<%= global_setting("default_pxe_item_global", "local") %>
```
Comment (iNecas) [inline provisioning_templates/PXEGrub2/pxegrub2_global_default.erb]: seems like index is also expected here.

File: provisioning_templates/PXEGrub2/pxegrub2_default_local_boot.erb (hunk 1 above)
Comment (iNecas) [reply provisioning_templates/PXEGrub2/pxegrub2_default_local_boot.erb:8]: Scratch that: I'm reviewing top-bottom, I see the changes in the snippets now :)

File: provisioning_templates/PXEGrub/pxegrub_global_default.erb (hunk 2 above)
Comment (lzap) [reply provisioning_templates/PXEGrub/pxegrub_global_default.erb]: I think a sane approach is to add `|| 0` which works with both PXELinux and Grub. Let me add that and rebase.

**Comment by ares** [comment]
core is merged, @iNecas were your comments addressed?

**Comment by iNecas** [comment]
Yes

**Comment by lzap** [comment]
What are we waiting for then? :-)

Docs: https://github.com/theforeman/theforeman.org/pull/1026

**Comment by mmoll** [comment]
merged, díky @lzap!

**Pull request merged**